File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/inheritance/baseDerivedWithMultipleConstructor/Dog.java
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.baseDerivedWithMultipleConstructor;

/**
 * Inherit the Animal class and can override makeNoise method
 * <p>
 * Note:
 * 1. Both the constructor calls the same super class constructor
 *
 * @author Ranjith Krishnamurthy
 */
public class Dog extends Animal {
    public Dog(String DNA) {
        super(DNA);
    }

    public Dog() {
        super("AAATGTG");
    }

    @Override
    public void makeNoise() {
        System.out.println("Bow Bow");
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/inheritance/baseDerivedWithMultipleConstructor/Dog.kt
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.baseDerivedWithMultipleConstructor

/**
 * Inherit the Animal class and can override makeNoise method
 *
 *
 * Note:
 * 1. Both the constructor calls the same super class constructor
 *
 * @author Ranjith Krishnamurthy
 */
class Dog : Animal {
    constructor(DNA: String) : super(DNA)

    constructor() : super("AAATGTG")

    override fun makeNoise() {
        println("Bow Bow")
    }
}